src/gleam_stdlib.erl

```diff
@@ -343,13 +343,17 @@ inspect(true) ->
 inspect(false) ->
     "False";
 inspect(Any) when is_atom(Any) ->
-    lists:map(
-        fun(Part) ->
-            [Head | Tail] = string:next_grapheme(unicode:characters_to_binary(Part)),
-            [string:uppercase([Head]), Tail]
-        end,
-        re:split(erlang:atom_to_list(Any), "_+", [{return, iodata}])
-    );
+    case re:run(Any, "^[a-zA-Z]+(_[a-zA-Z]+)*$", [unicode]) == nomatch of
```

Comment:
Let's do this by iterating over the string rather than using regex, it'll be faster. We could also do the work of splitting on `_` in that function too, to make it even faster.

Reply:
`Foo0` is a valid type in gleam right?
What does it translate to? `foo_0`?
`Foo01` transplates to `foo_0_1`?

Reply:
Yup! `One0` compiles to `one0`, `Two12` compiles to `two12`.